### PR TITLE
Remove a bunch of useless calls to count()

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Should improve performance in some cases.

WIP (for an unknown reason, tests don't pass locally, but it looks unrelated).

### CODE REVIEW

**Comment by sroze** [comment]
My personal taste is that `(0 === count($something))` is better than `($something)` as it describes better what is the logic and helps understand what kind of value you expect to be in such variable.

**Comment by dunglas** [comment]
The problem is that `count()` loops over the collection, so it's slower. If you really want to "describe", you should use `empty()` instead, but IMO it's not necessary (it's just more visual debt 😛).

File: src/Symfony/Component/Form/Form.php [hunk 1]
```diff
@@ -815,7 +815,7 @@ public function getErrors($deep = false, $flatten = true)
 
                 $iterator = $child->getErrors(true, $flatten);
 
-                if (0 === count($iterator)) {
+                if (!$iterator) {
```
Comment (chalasr) [inline src/Symfony/Component/Form/Form.php]: same, `$iterator` can be a `\Countable` (FormErrorIterator), this wouldn't be equivalent

**Comment by fabpot** [comment]
Not the kind of changes we do in 2.7. Should be moved to 3.4.

**Pull request closed without merging**